src/test/java/fi/aalto/cs/apluscourses/intellij/actions/CourseProjectActionTest.java

```diff
@@ -163,6 +160,7 @@ public void testCreateCourseProject() {
 
   @Test
   public void testNotifiesUserOfCourseInitializationError() {
+    createMockObjects();
```

Comment:
this is sure ugly, but it seems, that methods annotated with `@Before` are executed before the class is somehow properly instantiated (and as I've now added the `BasePlatformTestCase` which is needed to get the embedded action executed properly (to have the context)), so the project, that is a result of `getProject()` is `null`.

I'd be happy to receive advice/help on solving this, thus secondary task :sob:

Reply:
My guess is that `BasePlatformTestCase` overrides the basic functionality of JUnit in a way that `@Before` methods are not called.

If you're willing to get rid of `BasePlatformTestCase` (something that I'd gladly welcome 😉 but opinions vary, I know) I'd give the action an `ActionLauncher` object (define that interface somewhere) as a constructor parameter, so I could mock `ActionUtil.launc(...)` method but I think the approach you've taken here is ok.

Reply:
I think I've tried multiple approaches, like `@RunWith(JUnit4Runner.class)`, but no luck, so I'm not sure atm if `BasePlatformTestCase` is the reason, thus, because it uses JUnit3, it's the first candidate for the malfunction :disappointed: